src/key.rs

```diff
@@ -26,6 +31,147 @@ pub enum Integer {
     U128(u128),
 }
 
+/// An opaque floating-point type which has a total ordering.
+#[derive(Debug, Clone, Copy, Serialize, Deserialize)]
```

Comment:
Why does this need to impl `Serialize` / `Deserialize`?

Reply:
Oops, that's a vestige from when I was looking at whether I could implement the visit/serialize portions by deriving since it would be simpler for any user who wanted a custom float policy for whatever reason (reject NaNs maybe).